Remotion/Data/DomainObjects.UnitTests/Infrastructure/ObjectIDStringSerialization/ObjectIDStringSerializerTest.cs

```diff
@@ -277,6 +277,44 @@ public void Parse_WithInvalidClassID_ThrowsFormatException ()
               "Serialized ObjectID 'Arder|5d09030c-25c2-4735-b514-46333bd28ac8|System.Guid' is invalid: 'Arder' is not a valid class ID."));
     }
 
+    [TestCase("Official|5d09030c-25c2-4735-b514-46333bd28ac8|System.Guid, mscorlib", typeof(Guid))]
+    [TestCase("Official|5d09030c-25c2-4735-b514-46333bd28ac8|System.Guid, mscorlib, Version=4.0.0.0", typeof(Guid))]
+    [TestCase("Official|23|System.Int32, mscorlib", typeof(Int32))]
+    [TestCase("Official|23|System.Int32 , mscorlib , Version=4.0.0.0", typeof(Int32))]
+    [TestCase("Official|test|System.String, mscorlib", typeof(String))]
+    [TestCase("Official|test| System.String , mscorlib , Version=4.0.0.0", typeof(String))]
+    [Test]
+    public void Parse_ExplicitAssemblyName (string idString, Type type)
+    {
+#if NET
+      Assert.That(
+          () => ObjectIDStringSerializer.Instance.Parse(idString),
+          Throws.TypeOf<FormatException>().With.Message.EqualTo(
+              $"Serialized ObjectID '{idString}' is invalid: type '{idString[(idString.LastIndexOf('|') + 1)..]}' is not supported."));
```

Comment:
You are using a different string-cut syntax here than in the Parse_WithInvalidAssemblyName_ThrowsFormatException  test. Could you please help me understand the reasoning this choice since to me, it doesn't help with the readability?

Reply:
It is irrelevant now since that part is replaced anyway but I got an error that `System.Range` was not supported in .NET Framework so I changed the one case expecting an error if other cases were left. Well that did not work out as expected :P